### PR TITLE
Move to using building with use_edge_modules for vcpkg parity.

### PR DESCRIPTION
**Pull request by JeffMill**

### CODE REVIEW

File: scripts/install-deps.sh [hunk 1]
```diff
@@ -205,22 +205,72 @@ do_install_azure_iot_sdk() {
...
+        # mqtt is to be used (default: ON)
         "-Duse_mqtt:BOOL=ON"
+
+        #
+        # azure-shared-c-utility
+        #
+
+        # Use the out of the box UUID that comes with the SDK (default: OFF)
+        "-Duse_default_uuid:BOOL=ON"
+        # build WebSockets support (default: ON)
```
Comment (jw-msft) [inline scripts/install-deps.sh:267]: Nit: Could you move wsio to Transport section above after use_mqtt?

**Review by jw-msft — APPROVED** [review]
Approved with a comment.

**Comment by jw-msft** [comment]
Looks like following is failing in the Ubuntu pipeline for the PR:
https://dev.azure.com/azure-device-update/adu-linux-client/_build/results?buildId=11729&view=logs&j=7c6194e6-e836-5655-f2ac-2b6736913186&t=ab252b29-1ef0-5733-0abe-705e7550972b&l=284

[138/239] Linking CXX executable src/utils/d2c_messaging/tests/out-ubuntu1804-amd64/bin/d2c_messaging_unit_test
FAILED: src/utils/d2c_messaging/tests/out-ubuntu1804-amd64/bin/d2c_messaging_unit_test src/utils/d2c_messaging/tests/d2c_messaging_unit_test_tests-b858cb2.cmake /iot-hub-device-update/out-ubuntu1804-amd64/src/utils/d2c_messaging/tests/d2c_messaging_unit_test_tests-b858cb2.cmake 
: && /usr/bin/c++ -Werror -D_FILE_OFFSET_BITS=64 -fPIE -D_FORTIFY_SOURCE=2 -fstack-protector-strong -Wformat -Werror=format-security -fno-rtti -Os -DNDEBUG -pie -z relro -z now src/utils/d2c_messaging/tests/CMakeFiles/d2c_messaging_unit_test.dir/main.cpp.o src/utils/d2c_messaging/tests/CMakeFiles/d2c_messaging_unit_test.dir/d2c_messaging_ut.cpp.o -o src/utils/d2c_messaging/tests/out-ubuntu1804-amd64/bin/d2c_messaging_unit_test  src/communication_abstraction/libcommunication_abstraction.a  src/utils/d2c_messaging/libd2c_messaging.a  src/utils/retry_utils/libretry_utils.a  src/communication_abstraction/libcommunication_abstraction.a  /usr/local/lib/libiothub_client.a  /usr/local/lib/libumqtt.a  /usr/local/lib/libaziotsharedutil.a  -liothub_client_mqtt_transport  -lprov_auth_client  -lhsm_security_client  -luhttp  src/adu_types/libadu_types.a  /usr/local/lib/libparson.a  src/utils/c_utils/libc_utils.a  src/logging/zlog/libzlog.a  src/utils/system_utils/libsystem_utils.a  src/utils/c_utils/libc_utils.a  src/logging/zlog/libzlog.a  src/utils/system_utils/libsystem_utils.a  /usr/local/lib/libaziotsharedutil.a  /usr/local/lib/libumock_c.a  -lpthread  /usr/local/lib/libaziotsharedutil.a  /usr/local/lib/libaziotsharedutil.a  /usr/lib/x86_64-linux-gnu/libcurl.so  /usr/lib/x86_64-linux-gnu/libssl.so  /usr/lib/x86_64-linux-gnu/libcrypto.so  -lpthread  -lm  -lrt  -luuid && cd /iot-hub-device-update/out-ubuntu1804-amd64/src/utils/d2c_messaging/tests && /tmp/cmake-3.23.2-linux-x86_64/bin/cmake -D TEST_TARGET=d2c_messaging_unit_test -D TEST_EXECUTABLE=/iot-hub-device-update/out-ubuntu1804-amd64/src/utils/d2c_messaging/tests/out-ubuntu1804-amd64/bin/d2c_messaging_unit_test -D TEST_EXECUTOR= -D TEST_WORKING_DIR=/iot-hub-device-update/out-ubuntu1804-amd64/src/utils/d2c_messaging/tests -D TEST_SPEC= -D TEST_EXTRA_ARGS= -D TEST_PROPERTIES= -D TEST_PREFIX= -D TEST_SUFFIX= -D TEST_LIST=d2c_messaging_unit_test_TESTS -D CTEST_FILE=/iot-hub-device-update/out-ubuntu1804-amd64/src/utils/d2c_messaging/tests/d2c_messaging_unit_test_tests-b858cb2.cmake -P /usr/local/lib/cmake/Catch2/CatchAddTests.cmake
/usr/bin/ld: cannot find -lprov_auth_client
/usr/bin/ld: cannot find -lhsm_security_client
collect2: error: ld returned 1 exit status

**Review by jw-msft — APPROVED** [review]

File: scripts/install-deps.sh (hunk 1 above)
Comment (JeffMill) [reply scripts/install-deps.sh:267]: i copied what the azure sdk CMakeLists.txt has for ordering.

**Comment by JeffMill** [comment]
verified ubuntu1804-amd64 by changing ADO docker container.  All build and test passed.

**Pull request merged**